client/src/App.vue

```diff
@@ -87,56 +90,92 @@ import * as types from '@/store/types'
     'checkInOnExpire'
   ])
 })
-export default class App extends Vue {
+export default class App extends Mixins(SessionTimerMixin) {
   rollbackOnExpire: boolean
   checkInOnExpire: boolean
 
+  async resetAppState () {
+    // Redirect to the start
+    // Catch any errors, so we don't get errors like:
+    // Avoided redundant navigation to current location: "/"
+    this.$router.replace('/').catch(() => {})
+
+    // Display the tabs
+    await newRequestModule.resetAnalyzeName()
+    await newRequestModule.mutateName('')
+    await newRequestModule.mutateDisplayedComponent('Tabs')
```

Comment:
These are not all async methods. (This doesn't hurt but it's misleading.)

Async methods are those that can be blocked on something, like a network call.

Also on line 122 below.

Reply:
(FWIW, we often need async/await in the unit tests to let Vuetify do some things before we verify stuff. In my experience we haven't needed this in the actual code.)

Reply:
Actions are async in Vue and can be awaited eg. resetAnalyzeName. Probably shouldn't have awaited those mutations.

Reply:
Are you sure? I see some async actions but not "resetAnalyzeName"...?

Reply:
Whether you label them as async or not, an action is dispatched, and as result is inherently async.
Typescript / linting doesn't pick up on it...
https://medium.com/@bogna.ka/everyday-life-in-vue-world-vuex-actions-asynchronous-nature-ad154d246e4b

Reply:
Oh wow. Thanks for the link. It would be nice if Vue fixed their docs then!!!

OK, as a **future** exercise, action method declarations should be updated to return an explicit promise type, so a developer doesn't have to guess whether the return type is void, any, or async!